graylog2-web-interface/src/views/components/widgets/WidgetActionsMenu.tsx

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+import * as React from 'react';
+import { useState, useContext, useCallback } from 'react';
+import styled from 'styled-components';
+
+import ExportModal from 'views/components/export/ExportModal';
+import MoveWidgetToTab from 'views/logic/views/MoveWidgetToTab';
+import { loadDashboard } from 'views/logic/views/Actions';
+import { IconButton } from 'components/common';
+import { ViewManagementActions } from 'views/stores/ViewManagementStore';
+import WidgetPosition from 'views/logic/widgets/WidgetPosition';
+import { TitlesActions, TitleTypes } from 'views/stores/TitlesStore';
+import { ViewActions } from 'views/stores/ViewStore';
+import View from 'views/logic/views/View';
+import SearchActions from 'views/actions/SearchActions';
+import CopyWidgetToDashboard from 'views/logic/views/CopyWidgetToDashboard';
+import Search from 'views/logic/search/Search';
+import type { ViewStoreState } from 'views/stores/ViewStore';
+import IfSearch from 'views/components/search/IfSearch';
+import { MenuItem } from 'components/graylog';
+import { WidgetActions } from 'views/stores/WidgetStore';
+
+import ReplaySearchButton from './ReplaySearchButton';
+import ExtraWidgetActions from './ExtraWidgetActions';
+import CopyToDashboard from './CopyToDashboardForm';
+import MoveWidgetToTabModal from './MoveWidgetToTabModal';
+import WidgetActionDropdown from './WidgetActionDropdown';
+import WidgetHorizontalStretch from './WidgetHorizontalStretch';
+
+import IfInteractive from '../dashboard/IfInteractive';
+import IfDashboard from '../dashboard/IfDashboard';
+import WidgetFocusContext from '../contexts/WidgetFocusContext';
+import WidgetContext from '../contexts/WidgetContext';
+
+const Container = styled.div`
+  > * {
+    margin-right: 5px;
+
+    :last-child {
+      margin-right: 0;
+    }
+  }
+`;
+
+const _updateDashboardWithNewSearch = (dashboard: View, dashboardId: string) => ({ search: newSearch }) => {
```

Comment:
We can simplify this a bit. We can add the `Search` as a third parameter and use an arrow function in line 83.